Added new reference / Added Hardlink support / Performance considerations (#4092)
MicrosoftDocs/windowsserverdocs: WindowsServerDocs/storage/refs/refs-overview.md

- Deploying ReFS on Storage Spaces with shared SAS enclosures is suitable for hosting archival data and storing user documents.

> [!NOTE]
> Storage Spaces **supports** *local non-removable direct-attached* via BusTypes SATA, SAS, NVME, or attached via HBA (aka RAID controller in pass-through mode).

> > Please consider **limitations** on Cluster Shared Volumes in general, or CSVs used on Storage Spaces or Storage Spaces Direct (S2D). Refer chapter functionality.

Reviewer: Just like in my other comment, the phrase "Refer chapter functionality." uses incomplete English syntax. A better English phrasing would be:
```suggestion
> Please consider **limitations** on Cluster Shared Volumes in general, or CSVs used on Storage Spaces or Storage Spaces Direct (S2D). Refer to the functionality chapter.
```

> [!IMPORTANT]
> If you plan to use ReFS for CSV (Cluster Shared Volumes), please consider the limitations to pre-format your later CSV volumes with ReFS.

> > For CSV: NTFS should be used for traditional SANs. ReFS should be used on top of S2D. Refer chapter functionality.

Reviewer: Readability improvement suggestion:
```suggestion
> For CSV: NTFS should be used for traditional SANs. ReFS should be used on top of S2D. Refer to the functionality chapter.
```
A secondary phrasing could be: "Refer to the manual chapter concerning functionality."

> [!NOTE]
> Backup targets include the above supported configurations. Please contact application and storage array vendors for support details on Fiber Channel and iSCSI SANs. For SANs, if features such as thin provisioning, TRIM/UNMAP, or Offloaded Data Transfer (ODX) are required, NTFS must be used.

### Performance considerations

Pro:

> ReFS is an advanced filesystem over NTFS in many ways. It can greatly improve the performance in read / write operation scenarios. This is especially the case when comparing operations like creating Hyper-V virtual disks, snapshots or merging them. These performance benefits are not limited to Hyper-V though. 

Reviewer: The way I read the first sentence, is that it is meant to compare ReFS to NTFS and find ReFS to be better. In such comparisons, it is necessary to be more precise about the difference, like e.g. (for example):
```suggestion
ReFS is a superior filesystem to NTFS in many ways. It can greatly improve the performance in read / write operation scenarios. This is especially the case when comparing operations like creating Hyper-V virtual disks, snapshots or merging them. These performance benefits are not limited to Hyper-V though. 
```
Again, my suggestion may be imprecise, mostly due to my lack of knowledge about this software technology. I am mostly trying to convey that in a comparison between two systems, where one is better than the other (in one or more ways), the comparison needs to include a qualifier to say how one of them is better than the other. When using the preposition "over" in a comparison, you often need to rephrase the preceding text to say how the preference is made or how one of them takes precedence over the other.

Reviewer: Another way to phrase the first sentence could be:
```suggestion
ReFS is an advanced file system, superior to NTFS in many ways. It can greatly improve the performance in read / write operation scenarios. This is especially the case when comparing operations like creating Hyper-V virtual disks, snapshots or merging disks. These performance benefits are not limited to Hyper-V, though. 
```
I also threw in a suggestion for replacing "them", because "them" can be ambiguous here. Readability comma added near the end (natural speech). File System should most likely be separated and not in one word (in contrast to other germanic languages).

Based on real world applications, this also applies for other containered data structures that are similar to the technology of VHDX, such as Storage Spaces, especially when mirrored, or S2D which uses mirroring as per concept, as well as FSLogix containers. This also applies to Citrix PVS images, to name a 3rd party application scenario.

Con:

> While **supported**, there exist scenarios where ReFS performance may not meet the expectations, compared to NTFS formatted volumes.

Reviewer: ```suggestion
While **supported**, scenarios exist where ReFS performance may not meet the expectations, compared to NTFS formatted volumes.
```

This mostly applies to Basic Disks and volumes using ReFS without Storage Spaces / S2D, where volumes contain a high number of **small files and fast changing files**.
This is by design and is caused by the need for metadata handling and scrubbing. The performance in these scenarios *can* cause slow application responsiveness or even timeouts from time sensitive applications. Example: Lotus Notes .ini file(s). Other results may be unexpected low read and write speeds for random I/Os compared to NTFS in the same software and hardware configuration.

> A typical scenario that may apply, based on the underlying hardware configuration, are typically Windows User Profiles, Temporary Folders of browsers in high-density or multi-user enviroments, such as Microsoft Remote Desktop Host (RDSH), Windows Virtual Desktop (WVD), or 3rd party applications like Citrix User Profile Management (UPM).

Reviewer: ```suggestion
Typical scenarios that may apply, based on the underlying hardware configuration, are typically Windows User Profiles, Temporary Folders of browsers in high-density or multi-user environments, such as Microsoft Remote Desktop Host (RDSH), Windows Virtual Desktop (WVD), or 3rd party applications like Citrix User Profile Management (UPM).
```

| Hard links | Yes<sup>1</sup> | Yes |
| Object IDs | No | Yes |
| Offloaded Data Transfer (ODX) | No | Yes |
| Short names (SFN 8.3) | No | Yes |
| Extended attributes | No | Yes |
| Disk quotas | No | Yes |
| Bootable | No | Yes |
| Page file support | No | Yes |
| Supported on removable media | No | Yes |
1. This feature becomes available on **newly formatted ReFS 3.5 volumes only**
Prerequisites: Windows 10 Enterprise / Windows 10 for Workstations or Windows Server vNext, starting from build 19536 or later.

> Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest version, but will unlock the use of hardlinks. 

Reviewer: Hmm. I feel a bit uneasy with this phrasing and the use of the term "unlock" here. Looking at the how it currently reads, the meaning does not seem to be obvious, at least not to me. I would like to ask if you meant "break", "discontinue", "turn off", or "unlink", because the current phrasing looks like it means 'unlock' in the meaning of "unlock the next level" or "unlock a new feature" in the software. If you mean 'unlock' as in "turn off" or "break", I suggest using a term with a more direct and clear meaning, like for example:
```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest version, but will also break current hardlinks. 
```
My suggestion may very well be not precise enough, because I don't actually know exactly what happens to hardlinks when ReFS is upgraded from version 3.5 to a newer one.

Reviewer: Depending on the correct understanding of the hardlink feature and upgrading ReFS to a newer version, it could also be phrased like this:
```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest version, but will disable existing usage of hardlinks. 
```

Author: Hi @illfated thanks for spotting this. It should be 

```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest ReFS version, but will **not** make the hardlinks feature available.
```
